README.md

```diff
@@ -506,9 +510,76 @@ is a list of objects in [parameter object format](#parameter-object-format). `me
   }
 ]
 ```
-
+### eventUpdates
+*Optional* Object containing events that were updated after they were added to the AppMap. The name
+of each attribute is the `id` of the updated event. The value of the attribute is an `event` object
+that should be used in place of the original event.
+#### Example
+```
+{
+  ...
+  "events": [
+    ...
+    {
+      "event": "call",
+      "http_server_request": {
+        "headers": {
+          "host": "localhost:8080",
+          "user-agent": "curl/7.79.1",
+          "accept": "application/json"
+        },
+        "path_info": "/owners/1/pets/1/edit",
+        "protocol": "HTTP/1.1",
+        "request_method": "GET"
+      },
+      "id": 172,
+      "thread_id": 182
+    },
+    ...
+  ]
+  ...
+  "eventUpdates": {
+    "172": {
+      "event": "call",
+      "http_server_request": {
```

Comment:
Is this a deep merge with the original event 172, or more of a shallow merge?

Reply:
I guess it needs to be a deep merge, because `normalized_path_info` is the only changed field.

Can all the other (duplicative) fields be omitted?

Reply:
I think it would be more clear as an example if unnecessary fields are omitted from the example - for example, the _headers_ that are already in the original event.

Reply:
They're actually not merged. The object in `eventUpdates` is used in place of the one in `events`: https://github.com/applandinc/appmap/pull/30/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R516 . So, it would have been wrong to have omitted any of the fields in the example.

I initially considered specifying that the objects be merged. But, it's not clear to me why that would be necessary, and the Java agent doesn't need it, so I went with this.